### PR TITLE
Update to new atomic_lti gem

### PR DESCRIPTION
**Pull request by jbasdf**

### CODE REVIEW

File: app/views/shared/_default_client_settings.html.erb [hunk 1]
```diff
@@ -60,6 +60,13 @@
       settings[:launch_context] = lti.launch_context
       settings[:tool_consumer_instance_guid] = lti.tool_consumer_instance_guid
       settings[:tool_consumer_instance_name] = lti.tool_consumer_instance_name
+
+      if state_validation = request.env["atomic.validated.state_validation"]&.with_indifferent_access
+        settings[:state] = state_validation["state"]
+        settings[:ltiStorageParams] = state_validation["lti_storage_params"]
+        settings[:stateVerified] = state_validation["verified_by_cookie"]
+      end
```
Comment (mpetrowi) [inline app/views/shared/_default_client_settings.html.erb:68]: How about this:
```
settings[:state_validation] = atomic.validated.state_validation
```

and then later call `ltilaunch(window.DEFAULT_SETTINGS.state_valdation)`

File: client/apps/hello_world/app.jsx [hunk 2]
```diff
@@ -45,7 +47,13 @@ if (window.DEFAULT_JWT) { // Setup JWT refresh
 const mainApp =  document.getElementById('main-app');
 initResizeHandler(mainApp);
 
-ReactDOM.render(
-  <Root store={store} />,
-  mainApp,
-);
+ltiLaunch(window.DEFAULT_SETTINGS).then((valid) => {
```
Comment (mpetrowi) [inline client/apps/hello_world/app.jsx:50]: Do we need to handle a promise reject?

File: client/apps/hello_world/app.jsx [hunk 3]
```diff
@@ -45,7 +47,13 @@ if (window.DEFAULT_JWT) { // Setup JWT refresh
...
+  if (valid) {
+    ReactDOM.render(
+      <Root store={store} />,
+      mainApp,
+    );
+  } else {
+    document.body.innerHTML = 'Invalid request. Please reload the page.';
+  }
+});
```
Comment (mpetrowi) [inline client/apps/hello_world/app.jsx:59]: We could also make the validation into a react component.  Then error handling and accessing settings would just work.

File: app/views/shared/_default_client_settings.html.erb (hunk 1 above)
Comment (jbasdf) [reply app/views/shared/_default_client_settings.html.erb:68]: I made that change.

**Pull request merged**